src/tree_node.cpp

```diff
@@ -106,13 +106,17 @@ NodeStatus TreeNode::executeTick()
     if(!substituted)
     {
       using namespace std::chrono;
+
       auto t1 = steady_clock::now();
+      std::shared_ptr<void> timer(nullptr, [&](...) {
```

Comment:
```suggestion
      // trick to prevent the compile from reordering the order of execution. See #861 
      // This makes sure that the code is executed at the end of this scope
      std::shared_ptr<void> execute_later(nullptr, [&](...) {
```